src/polyfills/fetch.js

```diff
@@ -0,0 +1,307 @@
+
+    /** @const {!FetchResponseHeaders} */
+    this.headers = new FetchResponseHeaders(xhr);
+
+    /** @type {boolean} */
+    this.bodyUsed = false;
+
+    /** @type {?ReadableStream} */
+    this.body = null;
+  }
+
+  /**
+   * Create a copy of the response and return it.
+   * @return {!FetchResponse}
+   */
+  clone() {
+    dev().assert(!this.bodyUsed, 'Body already used');
+    return new FetchResponse(this.xhr_);
+  }
+
+  /**
+   * Drains the response and returns the text.
+   * @return {!Promise<string>}
+   * @private
+   */
+  drainText_() {
+    dev().assert(!this.bodyUsed, 'Body already used');
+    this.bodyUsed = true;
+    return Promise.resolve(this.xhr_.responseText);
+  }
+
+  /**
+   * Drains the response and returns a promise that resolves with the response
+   * text.
+   * @return {!Promise<string>}
+   */
+  text() {
+    return this.drainText_();
+  }
+
+  /**
+   * Drains the response and returns the JSON object.
+   * @return {!Promise<!JsonObject>}
+   */
+  json() {
+    return /** @type {!Promise<!JsonObject>} */ (
+      this.drainText_().then(parseJson));
+  }
+
+  /**
+   * Drains the response and returns a promise that resolves with the response
+   * ArrayBuffer.
+   * @return {!Promise<!ArrayBuffer>}
+   */
+  arrayBuffer() {
+    return /** @type {!Promise<!ArrayBuffer>} */ (
+      this.drainText_().then(utf8Encode));
+  }
+}
+
+/**
+ * Normalized method name by uppercasing.
+ * @param {string|undefined} method
+ * @return {string}
+ * @private
+ */
+function normalizeMethod(method) {
+  if (method === undefined) {
+    return 'GET';
+  }
+  method = method.toUpperCase();
+  dev().assert(
+      allowedMethods.includes(method),
+      'Only one of %s is currently allowed. Got %s',
+      allowedMethods.join(', '),
+      method
+  );
+  return method;
+}
+
+
+/**
+ * Provides access to the response headers as defined in the Fetch API.
+ * @private Visible for testing.
+ */
+class FetchResponseHeaders {
+  /**
+   * @param {!XMLHttpRequest|!XDomainRequest|!XMLHttpRequestDef} xhr
+   */
+  constructor(xhr) {
+    /** @private @const {!XMLHttpRequest|!XDomainRequest|!XMLHttpRequestDef} */
+    this.xhr_ = xhr;
+  }
+
+  /**
+   * @param {string} name
+   * @return {string}
+   */
+  get(name) {
+    return this.xhr_.getResponseHeader(name);
+  }
+
+  /**
+   * @param {string} name
+   * @return {boolean}
+   */
+  has(name) {
+    return this.xhr_.getResponseHeader(name) != null;
+  }
+}
+/**
+ * Returns instance of Response.
+ * @param {?ResponseBodyInit=} body
+ * @param {!ResponseInit=} init
+ */
+export function Response(body, init = {}) {
+  const lowercasedHeaders = map();
+  const data = Object.assign({
+    status: 200,
+    statusText: 'OK',
+    responseText: (body ? String(body) : ''),
+    /**
+     * @param {string} name
+     * @return {string}
+     */
+    getResponseHeader(name) {
+      return lowercasedHeaders[String(name).toLowerCase()] || null;
```

Comment:
`hasOwnProperty`, since the value could be an empty string.

Reply:
done